src/clj/fluree/db/query/fql/parse.cljc

```diff
@@ -683,7 +683,9 @@
                        :else            (where/match-iri id))]
     (reduce (partial parse-pred-cmp allowed-vars subj-cmp)
             triples
-            (dissoc node :id :idx))))
+            (->> (dissoc node :id :idx)
+                 ;; deterministic patterns for each pred
+                 (sort-by (comp str first))))))
```

Comment:
What is `(str (first node))` getting exactly? A string-ified key-value pair?

Reply:
A stringified key, actually. `sort-by` coerces the node map into a seq of entries `([k v] ...)`, so `first` gets the key and `str` stringifies it - I wanted to avoid any chance that keywords wouldn't be comparable with strings.